### ISSUE TITLE
`ServiceWorkerRegistration.update()` should return `Promise<ServiceWorkerRegistration>`

### ISSUE THREAD

**Issue by jespertheend** [microsoft/TypeScript-DOM-lib-generator#1684]
Currently it is defined in lib.dom.ts to return `Promise<void>`.

See the [MDN page for update()](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/update) which mentions:

> Return value
A [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that resolves with a [ServiceWorkerRegistration](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration) object.

[The spec](https://w3c.github.io/ServiceWorker/#service-worker-registration-update) is a bit unclear because it mentions some job logic:

> 6. Let job be the result of running [Create Job](https://w3c.github.io/ServiceWorker/#create-job) with update

[but in the end](https://w3c.github.io/ServiceWorker/#resolve-job-promise-algorithm) it seems like this promise is always resolved with a ServiceWorkerRegistration.

**Comment by skyclouds2001** [comment]
but the spec idl says this method return void, though in browser implement is seems to return with a ServiceWorkerRegistration

**Comment by jespertheend** [comment]
Hmm interesting, seems like [an issue was submitted](https://github.com/w3c/ServiceWorker/issues/1472) about this 5 years ago but it has never been addressed.

**Comment by skyclouds2001** [comment]
> Hmm interesting, seems like [an issue was submitted](https://github.com/w3c/ServiceWorker/issues/1472) about this 5 years ago but it has never been addressed.

Yes, I check the algorithm, and agree this should be a mistake in webidl

**Comment by Bashamega** [comment]
Hello @saschanaz 
Can I work on this issue?

**Comment by saschanaz** [comment]
No, this method returning an object is Chrome-only and other browser engines return `undefined` as the spec says.

If we want to change this then we should change the spec. Also commented on the spec issue.

**Comment by jespertheend** [comment]
The spec has now been updated with a fix.

Seems like this is [no longer an issue in TypeScript](https://www.typescriptlang.org/play/?ts=5.9.2#code/BQQwzgngdgxgBMAlHAvAPjgbwFBz3GAeyjABc4AnAUwHMBLMikUu41OEAdxDvKhABudGs0IUAdGCoUhMKgHUxAa2njq9MtOAAicQHowncQCsw2xAG5c+IiXLUwAVwA25FB269KtBqSYticUcABwATZiokK3wCYjBCZypxZ0IaYAcXUkkiYKpLbABfRCigA) anymore either since 5.9.2.
